redbot/core/commands/requires.py

```diff
@@ -326,13 +331,13 @@ def __init__(
 
     @staticmethod
     def get_decorator(
-        privilege_level: Optional[PrivilegeLevel], user_perms: Dict[str, bool]
+        privilege_level: Optional[PrivilegeLevel], user_perms: Optional[Dict[str, bool]]
     ) -> Callable[["_CommandOrCoro"], "_CommandOrCoro"]:
         if not user_perms:
             user_perms = None
 
         def decorator(func: "_CommandOrCoro") -> "_CommandOrCoro":
-            if asyncio.iscoroutinefunction(func):
+            if inspect.iscoroutinefunction(func):
```

Comment:
This will make decorators stop working with generator-based coros (`@asyncio.coroutine`).
I'm completely fine with this, especially that they're deprecated since Python 3.8 but I think this is technically also a breaking change which should be mentioned as part of the warning just in case.